modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -33,13 +35,13 @@ def do_remove(name, type)
       nil
     end
 
-    def execute(cmd, msg = nil, error_only = false)
+    def execute(*args, **keyword_args)
       tsecs = 5
       response = nil
       interpreter = Proxy::SETTINGS.x86_64 ? 'c:\windows\sysnative\cmd.exe' : 'c:\windows\system32\cmd.exe'
-      command = interpreter + ' /c c:\Windows\System32\dnscmd.exe ' + "#{@server} #{cmd}"
-
+      command = interpreter + ' /c c:\Windows\System32\dnscmd.exe ' + "#{@server}" + args.join(" ")
       std_in = std_out = std_err = nil
+      Open3.popen3(interpreter, *args)
```

Comment:
I believe this is incorrect, this belong to the line 46, this method returns array of three values.

For easier testing, I suggest you you create a private method named `popen3` so you can easily mock it without mocking the system method from Open3 module (this can sometimes be painful).